### PR TITLE
Delete posts, comments and messages

### PR DESCRIPTION
**Pull request by Sjmarf**
Adds a "delete" button for posts, comments and messages. Uses [this MlemMiddleware PR](https://github.com/mlemgroup/MlemMiddleware/pull/33).

### CODE REVIEW

File: Mlem/App/Utility/Extensions/Content Models/DeletableProviding+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            Task {
+                let result = await task.result.get()
+                switch result {
+                case .succeeded:
+                    ToastModel.main.add(
+                        .undoable(
+                            title: "Deleted",
+                            systemImage: Icons.deleteFill,
+                            callback: { self.updateDeleted(false) },
+                            color: .red
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Content Models/DeletableProviding+Extensions.swift]: *nit*

```suggestion
                            color: Palette.main.negative
```

File: Mlem/App/Utility/Extensions/Content Models/Message1Providing+Extensions.swift [hunk 2]
```diff
@@ -10,7 +10,7 @@ import MlemMiddleware
 extension Message1Providing {
     var self2: (any Message2Providing)? { self as? any Message2Providing }
     
-    var isOwnMessage: Bool { (AppState.main.firstAccount as? UserAccount)?.id == creatorId }
+    var isOwnMessage: Bool { creatorId == api.myPerson?.cacheId }
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Content Models/Message1Providing+Extensions.swift]: *nit*

Inconsistent with `Comment1Providing`:

`var isOwnComment: Bool { creatorId == api.myPerson?.id }`

File: Mlem/App/Utility/Extensions/Content Models/DeletableProviding+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+//
+
+import MlemMiddleware
+
+extension DeletableProviding {
+    func toggleDeleted(feedback: Set<FeedbackType>) {
+        if feedback.contains(.toast), !deleted {
+            let task = toggleDeleted()
+            Task {
+                let result = await task.result.get()
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Content Models/DeletableProviding+Extensions.swift]: Build failure on Xcode 15
```suggestion
                let result = try await task.result.get()
```

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Looks good!

File: Mlem/App/Utility/Extensions/Content Models/Message1Providing+Extensions.swift (hunk 2 above)
Comment (Sjmarf) [reply Mlem/App/Utility/Extensions/Content Models/Message1Providing+Extensions.swift]: Nice catch 👍

**Review by EricBAndrews — APPROVED** [review]
LGTM! 🚀

**Pull request merged**